lib/bundler/definition.rb

```diff
@@ -113,6 +114,10 @@ def initialize(lockfile, dependencies, sources, unlock, ruby_version = nil, opti
         @unlock[:gems] = @locked_specs.for(eager_unlock).map(&:name)
       end
 
+      if @unlock[:sources]
+        @unlock[:sources].each {|s| @sources.add_rubygems_remote(s) } if @unlock.is_a?(Hash) && @unlock[:sources]
```

Comment:
You already check `if unlock[:sources]` in the surrounding `if` block. The last bit of the suffix `if` can be removed, right?